index.js

```diff
@@ -20,11 +20,14 @@ module.exports = class extends URL {
      * @param string base Base URL
      */
     constructor(url, base) {
-        super(url, base);
-
-        // workaround for firefox bug
-        if (this.origin === "null") {
-            Object.defineProperty(this, "origin", {get: () => null});
+        try {
+            super(url, base);
+            Object.defineProperty(this, "isValid", { get: () => true });
```

Comment:
This is just a binary. We should be using `value: true, writable: false` IMO, as there's no actual getter logic involved.

Reply:
Makes sense, thanks!